daemon/src/setup_contract.rs

```diff
@@ -131,9 +135,9 @@ pub async fn new(
         .context("Failed to send Msg0")?;
     let msg0 = stream
         .select_next_some()
-        .timeout(Duration::from_secs(60))
+        .timeout(MSG_TIMEOUT)
         .await
-        .context("Expected Msg0 within 60 seconds")?
+        .context(format_expect_msg_within("Msg0"))?
```

Comment:
Note: This will allocate but only use the message in case of an error. To avoid the allocation in the happy path, you should use `.with_context` every time you need to allocate to construct the error message.

Reply:
thanks, I'll submit a small fixup for it!